src/Maestro/SubscriptionActorService/PullRequestActor.cs

```diff
@@ -1075,6 +1080,37 @@ private async Task<RepositoryBranchUpdate> GetRepositoryBranchUpdate()
 
         private static string GetNewBranchName(string targetBranch) => $"darc-{targetBranch}-{Guid.NewGuid()}";
 
+        private static async Task<List<(UpdateAssetsParameters update, List<DependencyUpdate> deps)>> ReplaceFromDependency(
+            IRemote darcRemote,
+            string targetRepository,
+            string targetBranch,
+            TargetRepoDependencyUpdate targetRepositoryUpdates
+            )
+        {
+            List<DependencyDetail> targetBranchDeps = (await darcRemote.GetDependenciesAsync(targetRepository, targetBranch)).ToList();
+
+            // 
+            var requiredDescriptionUpdates = targetRepositoryUpdates.RequiredUpdates
+                .Select(requiredUpdates =>
+                {
+                    requiredUpdates.deps = requiredUpdates.deps
+                        .Select(update =>
+                        {
+                            update.From = targetBranchDeps
+                                .Where(replace => update.From.Name == replace.Name)
+                                .FirstOrDefault(update.From);
+
+                            return update;
```

Comment:
I am wondering about one thing - this will effectively change the original object and create a duplicate collection of those. Are we fine with that?

Shouldn't we create a new object with a different From instead?

Otherwise we don't need to create new collections but we could just `foreach` over this and modify the From field.

Reply:
Oh, my understanding was that `.Select` operates on a copy of the elements in the collection and does not alter the original elements. If this is wrong as you point out then creating a new object is what I would agree with

Reply:
Select is just a `map()`. It operates on the elements of the original collection and then it depends if those are reference types.

Reply:
Should be creating a new `DependencyUpdate` object now